### ISSUE TITLE
Reason for removing files within *.esm/*.esp folder among loose files?

### ISSUE THREAD

**Issue by x43x61x69** [Lord-Akkrand/Skyrim-NX-Toolkit#49]
As title, I've noticed the tool by default will remove any folder named `*.esm/*.esp` within the meshes and texture folder (e.g. under facegendata/facegeom) in `pack_mod.py`.

Are those files unsupported? Many MODs were used to patch another existing MOD, which would require to have those file otherwise they won't do anything.

**Comment by Lord-Akkrand** [comment]
Doesn't it put them in a differently named folder, to match what's being
packed?  I'm sure I remember there being a reason for this, although I
honestly haven't looked at the code in a couple of years

**Comment by x43x61x69** [comment]
I’ve encountered a MOD that has 800MB of NIF and DDS but only within those .esp folders. After the conversation, it generated no BSA file and the log shown they were deleted even tho they were converted. I googled but didn’t find the exact reason why they were removed. If those .esp folders were for game default ones such as Skyrim.esm, that could make sense, but if it’s for another 3rd party MOD, removing them seems making no sense unless Switch version does not support such replacement.

**Comment by Lord-Akkrand** [comment]
Did you have the mods esp file in the same directory as the original BSA?
Or, if the mod came with loose files instead of a BSA, the esp file needs
to be in the root folder.  Otherwise, the packer doesn't see anything that
will load those files, so it deletes them.

**Comment by x43x61x69** [comment]
Yes I did, and I knew that they will be removed if there's no reference to them (however I do think there should be an option for the user to decide if they want to do simple packing instead, as for Switch it's much easier to pack loose files into BSAs for managing and load order between plugins). Here's [one of the examples](https://www.nexusmods.com/skyrimspecialedition/mods/595?tab=files), which use loose files. After converting, there'll be no BSA created as they were all in the `.esp/.esp` folders.

There are some other examples, where the original PC BSA has files within the BSA under the `.esp/.esm` folder and got completely removed after converted, cause the facegen data to be missing.

Thanks again for all your hard work.
